src/test/java/io/confluent/idesidecar/restapi/integration/LocalITs.java

```diff
@@ -0,0 +1,267 @@
+import io.confluent.idesidecar.restapi.kafkarest.RecordsV3ApiSuite;
+import io.confluent.idesidecar.restapi.kafkarest.api.ClusterV3Suite;
+import io.confluent.idesidecar.restapi.kafkarest.api.PartitionV3Suite;
+import io.confluent.idesidecar.restapi.kafkarest.api.TopicConfigV3Suite;
+import io.confluent.idesidecar.restapi.kafkarest.api.TopicV3Suite;
+import io.confluent.idesidecar.restapi.messageviewer.SimpleConsumerSuite;
+import io.confluent.idesidecar.restapi.resources.KafkaConsumeSuite;
+import io.confluent.idesidecar.restapi.testutil.NoAccessFilterProfile;
+import io.confluent.idesidecar.restapi.util.LocalTestEnvironment;
+import io.confluent.idesidecar.restapi.util.TestEnvironment;
+import io.quarkus.test.junit.QuarkusIntegrationTest;
+import io.quarkus.test.junit.TestProfile;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Tag;
+
+public class LocalITs {
+
+  private static final LocalTestEnvironment TEST_ENVIRONMENT = new LocalTestEnvironment();
+
+  @BeforeAll
+  static void beforeAll() {
+    // Start the test environment and use it for all tests
+    TEST_ENVIRONMENT.start();
+  }
+
+  @AfterAll
+  static void afterAll() {
+    // Shutdown the test environment after all tests have run
+    TEST_ENVIRONMENT.shutdown();
+  }
+
+  @Nested
+  class LocalConnectionTests {
+
+    @QuarkusIntegrationTest
+    @Tag("io.confluent.common.utils.IntegrationTest")
+    @TestProfile(NoAccessFilterProfile.class)
+    @Nested
+    class RecordTests extends AbstractIT implements RecordsV3ApiSuite {
+
+      @Override
+      public TestEnvironment environment() {
+        return TEST_ENVIRONMENT;
+      }
+
+      @BeforeEach
+      @Override
+      public void setupConnection() {
+        setupConnection(this, TestEnvironment::localConnectionSpec);
+      }
+    }
+
+    @QuarkusIntegrationTest
+    @Tag("io.confluent.common.utils.IntegrationTest")
+    @TestProfile(NoAccessFilterProfile.class)
+    @Nested
+    class RecordFailureTests extends AbstractIT implements RecordsV3ApiErrorsSuite {
+
+      @Override
+      public TestEnvironment environment() {
+        return TEST_ENVIRONMENT;
+      }
+
+      @BeforeEach
+      @Override
+      public void setupConnection() {
+        setupConnection(this, TestEnvironment::localConnectionSpec);
+      }
+    }
+
+    @QuarkusIntegrationTest
+    @Tag("io.confluent.common.utils.IntegrationTest")
+    @TestProfile(NoAccessFilterProfile.class)
+    @Nested
+    class TopicTests extends AbstractIT implements TopicV3Suite, PartitionV3Suite {
+
+      @Override
+      public TestEnvironment environment() {
+        return TEST_ENVIRONMENT;
+      }
+
+      @BeforeEach
+      @Override
+      public void setupConnection() {
+        setupConnection(this, TestEnvironment::localConnectionSpec);
+      }
+    }
```

Comment:
A `@Nested` test class exists for each kind of connection to be tested: the `LocalConnectionTests` here is for local connections. This class could have implemented all of the IT suite interfaces, but that would have put a hundred tests (including dynamic tests) all on this one class -- the test results were a bit hard to navigate.

So instead, we define multiple `@Nested` (2-levels deep) test classes that implement a subset of the _IT suite_ interfaces. This organizes them a bit more nicely. For example, above the `RecordTests` implements the `RecordsV3ApiSuite` interface and thus gets all of its test methods and runs them against a _local_ connection that is reused for all of the tests in `LocalConnectionTests`.

Likewise, there is a `RecordFailuresTest` that implements the `RecordsV3ApiErrorsSuite` tests, a `TopicsTest` that implements the `TopicV3Suite` and `PartitionV3Suite` tests, etc.

Lower down in this file is a top-level `@Nested` test class called `DirectConnectionWithoutCredentialsTests`, with nested test classes that implement various test suites, that runs *the same tests* against a direct connection with no credentials using the `LocalTestEnvironment`'s containers.